packages/tx/src/EIP2930Transaction.ts

```diff
@@ -0,0 +1,391 @@
+    const cost = super.getDataFee()
+    const accessListStorageKeyCost = this.common.param('gasPrices', 'accessListStorageKeyCost')
+    const accessListAddressCost = this.common.param('gasPrices', 'accessListAddressCost')
+
+    let slots = 0
+    for (let index = 0; index < this.accessList.length; index++) {
+      const item = this.accessList[index]
+      const storageSlots = item[1]
+      slots += storageSlots.length
+    }
+
+    const addresses = this.accessList.length
+    cost.iaddn(addresses * accessListAddressCost + slots * accessListStorageKeyCost)
+    return cost
+  }
+
+  /**
+   * Returns a Buffer Array of the raw Buffers of this transaction, in order.
+   * @param asList - By default, this method returns a concatenated Buffer
+   *                 If this is not desired, then set this to `true`, to get a Buffer array.
+   */
+  raw(asList = false): Buffer[] | Buffer {
+    const base = <Buffer[]>[
+      bnToRlp(this.chainId),
+      bnToRlp(this.nonce),
+      bnToRlp(this.gasPrice),
+      bnToRlp(this.gasLimit),
+      this.to !== undefined ? this.to.buf : Buffer.from([]),
+      bnToRlp(this.value),
+      this.data,
+      this.accessList,
+      this.v !== undefined ? bnToRlp(this.v) : Buffer.from([]),
+      this.r !== undefined ? bnToRlp(this.r) : Buffer.from([]),
+      this.s !== undefined ? bnToRlp(this.s) : Buffer.from([]),
+    ]
+    if (!asList) {
+      return Buffer.concat([Buffer.from('01', 'hex'), rlp.encode(base)])
+    } else {
+      return base
+    }
+  }
+
+  /**
+   * Returns the encoding of the transaction. For typed transaction, this is the raw Buffer.
+   * In LegacyTransaction, this is a Buffer array.
+   */
+  serialize(): Buffer {
+    return <Buffer>this.raw()
+  }
+
+  /**
+   * Returns an object with the JSON representation of the transaction
+   */
+  toJSON(): JsonTx {
+    const accessListJSON = []
+
+    for (let index = 0; index < this.accessList.length; index++) {
+      const item: any = this.accessList[index]
+      const JSONItem: any = {
+        address: '0x' + setLengthLeft(<Buffer>item[0], 20).toString('hex'),
+        storageKeys: [],
+      }
+      const storageSlots: Buffer[] = item[1]
+      for (let slot = 0; slot < storageSlots.length; slot++) {
+        const storageSlot = storageSlots[slot]
+        JSONItem.storageKeys.push('0x' + setLengthLeft(storageSlot, 32).toString('hex'))
+      }
+      accessListJSON.push(JSONItem)
+    }
+
+    return {
+      chainId: bnToHex(this.chainId),
+      nonce: bnToHex(this.nonce),
+      gasPrice: bnToHex(this.gasPrice),
+      gasLimit: bnToHex(this.gasLimit),
+      to: this.to !== undefined ? this.to.toString() : undefined,
+      value: bnToHex(this.value),
+      data: '0x' + this.data.toString('hex'),
+      accessList: accessListJSON,
+    }
+  }
+
+  public isSigned(): boolean {
+    const { yParity, r, s } = this
+    return yParity !== undefined && !!r && !!s
+  }
+
+  public hash(): Buffer {
+    // TODO add decorator
```

Comment:
if this comment is for `!this.isSigned()`, unfortunately decorators can't be dynamic